ignite/contrib/engines/common.py

```diff
@@ -575,7 +598,7 @@ def gen_save_best_models_by_val_score(
         to_save = {"model": models}
 
     best_model_handler = Checkpoint(
-        to_save,
+        cast(Dict[str, torch.nn.Module], to_save),
```

Comment:
No need to use `cast` here, you can adjust the code
https://github.com/pytorch/ignite/blob/92455594401681bbd3e675a664072b905401274b/ignite/contrib/engines/common.py#L596-L598
to
```python
    if isinstance(models, nn.Module):
        to_save = {"model": models}  # type: Dict[str, nn.Module]
    else:
        to_save = models
```